carbon-design-system/carbon
# feat(numberinput): add support for "any" value for step prop

Closes #6930 

Currently Carbon does not allow for string values for the `step` prop. This is preventing users from using "any" as a value (without proptype warning) which in turn triggers validation when input value is submitted. This PR allows for both numbers and strings. 

#### Changelog

**New**


- added typeof check to see if value was a string and prevent rendering or step logic if so in NumberInput.js
- added test to ensure that step buttons were not rendered when the value of "any" was supplied

**Changed**

- changed proptype declaration of `step` to `oneOfType` in NumberInput.js
- changed NumberInput story to showcase the use of this prop value



#### Testing / Reviewing

- go to NumberInput story
- change knob to any of the different options
- see that buttons go away and no error message is present on hover when adding decimal to number input
